doc/user/content/transform-data/troubleshooting.md

```diff
@@ -277,6 +277,317 @@ If you've gone through the dataflow troubleshooting and do not want to make any
+
+To show how you can use
+[`mz_introspection.mz_lir_mapping`](/sql/system-catalog/mz_introspection/#mz_lir_mapping)
+to attribute performance characteristics, the attribution examples in this
+section reference the `wins_by_item` index and the `winning_bids` view from the
+[quickstart guide](/get-started/quickstart/#step-2-create-the-source):
+
+```sql
+CREATE SOURCE auction_house
+FROM LOAD GENERATOR AUCTION
+(TICK INTERVAL '1s', AS OF 100000)
+FOR ALL TABLES;
+
+CREATE VIEW winning_bids AS
+  SELECT DISTINCT ON (a.id) b.*, a.item, a.seller
+    FROM auctions AS a
+    JOIN bids AS b
+      ON a.id = b.auction_id
+   WHERE b.bid_time < a.end_time
+     AND mz_now() >= a.end_time
+   ORDER BY a.id, b.amount DESC, b.bid_time, b.buyer;
+
+CREATE INDEX wins_by_item ON winning_bids (item);
+```
+
+We attribute four different kinds of performance data to parts of the
+`wins_by_item` query:
+
+- [computation time](#attributing-computation-time)
+- [memory usage](#attributing-memory-usage)
+- [Top-k query sizing hints](#attributing-topk-hints)
+- [worker skew](#localizing-worker-skew)
+
+### Attributing computation time
+
+When optimizing a query, it helps to be able to attribute 'cost' to its parts,
+starting with how much time is spent computing in each part overall. Materialize
+reports the time spent in each _dataflow operator_ in
+[`mz_introspection.mz_compute_operator_durations_histogram`](/sql/system-catalog/mz_introspection/#mz_compute_operator_durations_histogram).
+By joining it with
+[`mz_introspection.mz_lir_mapping`](/sql/system-catalog/mz_introspection/#mz_lir_mapping),
+we can attribute the time spent in each operator to the higher-level, more
+intelligible LIR operators.
+
+For example, to find out how much time is spent in each operator for the `wins_by_item` index and the `winning_bids` view, run the following query:
+
+```sql
+SELECT mo.name AS name, global_id, lir_id, parent_lir_id, REPEAT(' ', nesting * 2) || operator AS operator,
+       SUM(duration_ns)/1000 * '1 microsecond'::INTERVAL AS duration, SUM(count) AS count
+    FROM           mz_introspection.mz_lir_mapping mlm
+         LEFT JOIN mz_introspection.mz_compute_operator_durations_histogram mcodh
+                ON (mlm.operator_id_start <= mcodh.id AND mcodh.id < mlm.operator_id_end)
+              JOIN mz_catalog.mz_objects mo
+                ON (mlm.global_id = mo.id)
+   WHERE mo.name IN ('wins_by_item', 'winning_bids')
+GROUP BY mo.name, global_id, lir_id, operator, parent_lir_id, nesting
+ORDER BY global_id, lir_id DESC;
+```
+
+The query produces results similar to the following (your specific numbers will
+vary):
+
+{{< yaml-table data="query_attribution_computation_time_output" >}}
+
+- The `duration` column shows that the `TopK` operator is where we spend the
+  bulk of the query's computation time.
+
+- Creating an index on a view executes the underlying view query. As such, the
```

Comment:
Wasn't sure if this is why we have the 2 global_ids, but ... wanted some explanation since we just state the index has the 2 ... and people might think it's just because of our `WHERE` clause.

Reply:
I like the changes; I replaced 'executed' with 'started', since 'execution' feels like it implies termination.

What else do we need here to be ready to go?

Reply:
lgtm.